tenseal/cpp/tensors/utils/matrix_ops.h

```diff
@@ -122,9 +121,18 @@ Ciphertext diagonal_ct_vector_matmul(shared_ptr<TenSEALContext> tenseal_context,
             std::min((i + 1) * batch_size, vector_size)));
     }
 
+    std::optional<std::exception> fail;
     for (size_t i = 0; i < n_jobs; i++) {
-        tenseal_context->evaluator->add_inplace(result,
-                                                future_results[i].get());
+        try {
+            tenseal_context->evaluator->add_inplace(result,
+                                                    future_results[i].get());
```

Comment:
My understanding is that when an asynchronous operation throws an exception, it will be propagated back to a thread that calls std::future::get().

It is mentioned here https://en.cppreference.com/w/cpp/thread/future/get
```
If an exception was stored in the shared state referenced by the future (e.g. via a call to std::promise::set_exception()) then that exception will be thrown.
```

So, the future will store the exception from the workers, and it will not throw it somewhere else than the get call.

Reply:
Got it, thanks for the explanation!